python/foglamp/services/core/snapshot.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+""" Provides utility functions to take snapshot of Foglamp"""
+
+import datetime
+import os
+from os.path import basename
+import glob
+import json
+import tarfile
+import fnmatch
+import time
+
+from foglamp.common import logger
+from foglamp.common.common import _FOGLAMP_ROOT
+
+
+__author__ = "Amarendra K Sinha"
+__copyright__ = "Copyright (c) 2017 OSIsoft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+_LOGGER = logger.setup(__name__)
+_NO_OF_FILES_TO_RETAIN = 3
```

Comment:
why?  I can imagine some limit but not sure if that is 3? /cc @MarkRiddoch

Reply:
This was not part of the original requirement, but is a good idea. The intended use is that there will probably only be a single snapshot, so 3 is as good a choice as any currently. Let's leave it at this, we should clearly indicate when we purge old ones in the log so that it is obvious if/when we get any issues form this.